src/vms/pvm/txs/fee/utils.ts

```diff
@@ -0,0 +1,94 @@
+import { Bytes } from '../../../../serializable';
+import { ID_LEN } from '../../../../serializable/fxs/common/id';
+import { INT_LEN } from '../../../../serializable/primitives/int';
+import { bufferToNumber } from '../../../../utils';
+
+// TODO: Why is there this offset on the message? What are the first two bytes?
+// Offset the start of deserialization because of some unknown bytes at the front.
+const START_INDEX = 2;
+// Offset for the WarpSignature type ID
+const SIGNATURE_INDEX_OFFSET = 4;
+
+/**
+ * Calculates the number of `1`s (set bits) in the binary representation of a number.
+ *
+ * @param num
+ * @returns The number of bits set to 1 in the binary representation of `num`
+ */
+const hammingWeight = (num: number): number => {
+  let count = 0;
+  while (num !== 0) {
+    count += num & 1;
+    num >>= 1;
+  }
+  return count;
+};
+
+/**
+ * Takes a big-endian byte slice and returns the number of bits set to 1.
+ *
+ * This function iterates through each byte in the `Uint8Array` and calculates
+ * the Hamming weight for each byte, summing up the results.
+ *
+ * @param bytes big-endian byte slice
+ * @returns number of bits set to 1
+ */
+export const bitsFromBytesLength = (bytes: Uint8Array): number => {
```

Comment:
this is a weird function name. should we also call this `hammingWeight`? does the endianness of the byte slice matter for this calculation? the comment seems to indicate it might